File: first_significant_digit_of_the_decimal_expansion_of_n_divided_by_2_to_the_power_of_n/A364881.cs
class A364881  {
    // Can produce the first 1023 terms.
    public static int aFirstMethod(int n) {
        double x = n/Math.Pow(2, n);
        return (int)Math.Floor(x / Math.Pow(10, Math.Floor(Math.Log10(x))));
    }
    // Can produce the first 1023 terms.
    public static string aSecondMethod(int n) {
        return string.Format("{0:e}", ((double)n)/Math.Pow(2, n)).Substring(0, 1);
    }

    public static int getSequenceSize(String methodName) {
        while(true) {
            try {
                Console.Write("Enter sequence size between [1-1023] for " + methodName + ": ");
                int n = Convert.ToInt32(Console.ReadLine());
                if(!(n >= 1 && n <= 1023)) {
                    throw new ArgumentOutOfRangeException();
                }
                return n;
            } catch (ArgumentOutOfRangeException) {
                Console.WriteLine("The number is not between [1-1023]!");
            } catch (Exception) {
                Console.WriteLine("Invalid input. Please enter a valid integer!");
            }
        }
    }
}

class TestDrive {
    static void Main(String[] args) {
        
        int sizeForFirstList = A364881.getSequenceSize("aFirstMethod");
        int[] listForFirstMethod = new int[sizeForFirstList];
        int sizeForSecondList = A364881.getSequenceSize("aSecondMethod");
        string[] listForSecondMethod = new string[sizeForSecondList];
        
        for (int n = 1; n <= sizeForFirstList; n++) {
            listForFirstMethod[n-1] = A364881.aFirstMethod(n);
        }
        for (int n = 1; n <= sizeForSecondList; n++) {
            listForSecondMethod[n-1] = A364881.aSecondMethod(n);
        }

        Console.WriteLine("results for first method:");
        Console.WriteLine(string.Join(", ", listForFirstMethod));
        Console.WriteLine("results for second method:");
        Console.WriteLine(string.Join(", ", listForSecondMethod));
    }
}

File: first_significant_digit_of_the_decimal_expansion_of_n_divided_by_2_to_the_power_of_n/A364881.cpp
#include <iostream>
#include <cmath>
#include <sstream>

using namespace std;
class A364881 {
    // Both methods can be used to generate the sequence.
    public:
        // Can produce the first 1023 terms.
        static int aFirstMethod(int n) {
            double x = n/pow(2, n);
            return floor(x / pow(10, floor(log10(x))));
        }

        // Can produce the first 1023 terms.
        static char aSecondMethod(int n) {
            double x = n/pow(2, n);
            ostringstream oss;
            oss << scientific << x;
            return oss.str()[0];
        }

        static int getSequenceSize(string methodName) {
            while (true) {
                try {
                    int n;
                    cout << "Enter sequence size between [1-1023] for " << methodName << ": ";
                    if(!(std::cin >> n)) {
                        cin.clear();
                        cin.ignore(numeric_limits<streamsize>::max(), '\n');
                        throw std::invalid_argument("Invalid input. Please enter a valid integer!");
                    } else if (!(n >= 1 && n <= 1023)) {
                        throw out_of_range("The number is not between [1-1023]!");
                    }
                    return n;
                }
                catch(const out_of_range& oor) {
                    cerr << oor.what() << endl;
                }
                catch(const exception& e) {
                    std::cerr << e.what() << '\n';
                }
            }
        }
};

// test drive
int main() {
    int sizeForFirstList = A364881::getSequenceSize("aFirstMethod");
    int listForFirstMethod[sizeForFirstList];
    int sizeForSecondList = A364881::getSequenceSize("aSecondMethod");
    char listForSecondMethod[sizeForSecondList];

    for (int n = 1; n <= sizeForFirstList; n++) {
        listForFirstMethod[n-1] = A364881::aFirstMethod(n);
    }
    for (int n = 1; n <= sizeForSecondList; n++) {
        listForSecondMethod[n-1] = A364881::aSecondMethod(n);
    }
    
    cout << "results for first method:" ;
    for(int n : listForFirstMethod) cout << n << " ";
    cout << endl;
    cout << "results for second method:" ;
    for(char n : listForSecondMethod) cout << n << " ";

    return 0;
}